pkg/aggregator/aggregator.go

```diff
@@ -257,3 +267,72 @@ func mergeSpecs(dest, source *spec.Swagger, renameModelConflicts, ignorePathConf
 	}
 	return nil
 }
+
+// deepEqualDefinitionsModuloGVKs compares a and b, but ignores the x-kubernetes-group-version-kind extension.
+// It will mutate a temporarily, but not b.
+func deepEqualDefinitionsModuloGVKs(a, b *spec.Schema) bool {
+	old, oldFound := a.ExtraProps["x-kubernetes-group-version-kind"]
+
+	if _, found := b.ExtraProps["x-kubernetes-group-version-kind"]; found {
+		a.ExtraProps["x-kubernetes-group-version-kind"] = b.ExtraProps["x-kubernetes-group-version-kind"]
+	} else {
+		delete(a.ExtraProps, "x-kubernetes-group-version-kind")
+	}
+	defer func() {
+		if oldFound {
+			a.ExtraProps["x-kubernetes-group-version-kind"] = old
+		} else {
+			delete(a.ExtraProps, "x-kubernetes-group-version-kind")
+		}
+	}()
+
+	return reflect.DeepEqual(a, b)
+}
+
+// mergedGVKs merges the x-kubernetes-group-version-kind slices and returns the result, and whether
+// x-kubernetes-group-version-kind was set at all in either v2 or v1.
+func mergedGVKs(s1, s2 *spec.Schema) (interface{}, bool, error) {
+	gvk1, found1 := s1.ExtraProps["x-kubernetes-group-version-kind"]
+	gvk2, found2 := s2.ExtraProps["x-kubernetes-group-version-kind"]
+
+	if !found1 {
+		return gvk2, found2, nil
+	}
+	if !found2 {
+		return gvk1, found1, nil
+	}
+
+	slice1, ok := gvk1.([]interface{})
+	if !ok {
+		return nil, false, fmt.Errorf("expected slice of GroupVersionKinds, got: %+v", slice1)
+	}
+	slice2, ok := gvk2.([]interface{})
+	if !ok {
+		return nil, false, fmt.Errorf("expected slice of GroupVersionKinds, got: %+v", slice2)
+	}
+
+	ret := make([]interface{}, len(slice1), len(slice1)+len(slice2))
+	copy(ret, slice1)
+	seen := make(map[string]bool, len(slice1)+len(slice2))
+	for _, x := range slice1 {
+		gvk, ok := x.(map[string]interface{})
+		if !ok {
+			return nil, false, fmt.Errorf(`expected {"group": <group>, "kind": <kind>, "version": <version>}, got: %#v`, x)
+		}
+		k := fmt.Sprintf("%s/%s.%s", gvk["group"], gvk["version"], gvk["kind"])
+		seen[k] = true
+	}
+	for _, x := range slice2 {
+		gvk, ok := x.(map[string]interface{})
+		if !ok {
+			return nil, false, fmt.Errorf(`expected {"group": <group>, "kind": <kind>, "version": <version>}, got: %#v`, x)
+		}
+		k := fmt.Sprintf("%s/%s.%s", gvk["group"], gvk["version"], gvk["kind"])
+		if seen[k] {
+			continue
+		}
+		ret = append(ret, x)
```

Comment:
nit: I would return true if this line at least ran once, i.e. we actually appended anything. this can also be done by checking the length of ret.

Reply:
good idea. Fixed.